src/main/java/org/springframework/samples/petclinic/rest/importcsv/ImportCSV.java

```diff
@@ -23,6 +23,109 @@
 @RequestMapping("api/import")
 public class ImportCSV {
 
+    class ParsedEntity {
+
+        private String field;
+        private int pos;
+
+        /***
+         * contructor
+         * @param field
+         * @param pos
+         */
+        public ParsedEntity(String field, int pos) {
+            this.field = field;
+            this.pos = pos;
+        }
+
+        public String getField() {
+            return field;
+        }
+
+        public void setField(String field) {
+            this.field = field;
+        }
+
+        public int getPos() {
+            return pos;
+        }
+
+        public void setPos(int pos) {
+            this.pos = pos;
+        }
+    }
+
+    /***
+     * parse the entity
+     * @param csv : csv string
+     * @param pos : start pared position
+     * @return
+     */
+    public ParsedEntity parseEntity(String csv, int pos) {
+        String field = "";
+
+        while (pos < csv.length() && (csv.charAt(pos) != ';' && csv.charAt(pos) != '\n')) {
+            field += csv.charAt(pos++);
+        }
+        return new ParsedEntity(field, pos);
+    }
+
+    /***
+     * delete the pet record
+     * @param pet
+     * @return
+     */
+    public boolean deregisterPetFromOwner(Pet pet) {
```

Comment:
I like that you kept the indentations small throughout the class :) except maybe in this class, the for-loop and three if-clauses could be eased a bit